src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp

```diff
@@ -0,0 +1,273 @@
+                                  });
+          feature_map.erase(map_it, feature_map.end());
+  
+          fm_info.feature_maps.push_back(feature_map);
+          fm_info.kd_tree.addMaps(fm_info.feature_maps); // KDTree references into feature_map
+  
+          // mapping of MS2 spectra to features
+          feature_mapping = FeatureMapping::assignMS2IndexToFeature(spectra,
+                                                                    fm_info,
+                                                                    precursor_mz_tol,
+                                                                    precursor_rt_tol,
+                                                                    precursorMzToleranceUnitIsPPM());
+        }
+        else
+        {
+          throw OpenMS::Exception::FileEmpty(__FILE__,
+                                             __LINE__,
+                                             __FUNCTION__,
+                                             "Error: FeatureXML was empty, please provide a valid file.");
+        }
+      }
+    }   
+
+    void SiriusExportAlgorithm::logFeatureSpectraNumber(const String& featureinfo,
+                                                         const FeatureMapping::FeatureToMs2Indices& feature_mapping,
+                                                         const MSExperiment& spectra) const
+    {
+      // number of features to be processed
+      if (isFeatureOnly() && !featureinfo.empty())
+      {
+        OPENMS_LOG_INFO << "Number of features to be processed: " << feature_mapping.assignedMS2.size() << std::endl;
+      }
+      else if (!featureinfo.empty())
+      {
+        OPENMS_LOG_INFO << "Number of features to be processed: " << feature_mapping.assignedMS2.size() << std::endl;
+        OPENMS_LOG_INFO << "Number of additional MS2 spectra to be processed: " << feature_mapping.unassignedMS2.size() << std::endl;
+      } 
+      else
+      {
+        long count_ms2 = count_if(spectra.begin(), spectra.end(),
+                [](const MSSpectrum &spectrum) { return spectrum.getMSLevel() == 2; });
+
+        OPENMS_LOG_INFO << "Number of MS2 spectra to be processed: " << count_ms2 << std::endl;
+      }
+    }
+
+    void SiriusExportAlgorithm::run(const StringList& mzML_files,
+                                    const StringList& featureXML_files,
+                                    const String& out_ms,
+                                    const String& out_compoundinfo) const
+    {
+      // loop over all spectra in all files and write data to ofstream
+      ofstream os;
+
+      // create temporary input file (.ms)
+      os.open(out_ms);
+      if (!os)
+      {
+        throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, out_ms);
+      }
+      os.precision(12);
+
+      std::vector<SiriusMSFile::CompoundInfo> v_cmpinfo; // To store compound information for all files
+      for (size_t i = 0; i < mzML_files.size(); ++i) {
```

Comment:
the formatting seems to be a bit of in this file in general. e.g.
```suggestion
      for (size_t i = 0; i < mzML_files.size(); ++i) 
      {
```
or tab sizes